### PR TITLE
core, dao: merge new update map for account and nft

### PR DESCRIPTION
**Pull request by KeefeL**
### Description

1. Merge the pending new account and update account.
2. Merge the pending new nft and update nft.

### Rationale

There is no need to use two different maps for the new and update operations, using one we can realize the same function, and the logic is simpler.

### Example

NA

### Changes

Notable changes:
* Merge new and update map for account and nft.

### CODE REVIEW

File: core/statedb/statedb.go [hunk 1]
```diff
@@ -193,6 +193,50 @@ func (s *StateDB) GetAccount(accountIndex int64) (*account.Account, error) {
 	return account, nil
 }
 
+func (s *StateDB) GetAccountByName(accountName string) (*account.Account, error) {
+	for accountIndex := range s.PendingAccountMap {
```
Comment (lightning-li) [inline core/statedb/statedb.go:197]: `s.PendingAccountMap` is same as `s.StateCache.PendingAccountMap`, so `exist` must be true. We can remove `GetPendingAccount` call

File: core/statedb/statedb.go [hunk 2]
```diff
@@ -553,16 +546,12 @@ func (s *StateDB) GetNextAccountIndex() int64 {
...
-		}
-		return maxNftIndex + 1
+	maxNftIndex, err := s.chainDb.L2NftModel.GetLatestNftIndex()
+	if err != nil {
+		panic("get latest nft index error: " + err.Error())
 	}
 
-	maxNftIndex := int64(-1)
-	for index := range s.PendingNewNftMap {
+	for index := range s.PendingNftMap {
```
Comment (guagualvcha) [inline core/statedb/statedb.go:554]: So is here, not performance friendly.

**Review by lightning-li — APPROVED** [review]

File: core/executor/deposit_executor.go [hunk 3]
```diff
@@ -41,31 +41,9 @@ func (e *DepositExecutor) Prepare() error {
...
-				account = tempAccount
-				exist = true
-				break
-			}
-		}
-
-		if !exist {
-			return errors.New("invalid account name hash")
-		}
+		return err
```
Comment (guagualvcha) [inline core/executor/deposit_executor.go:46]: Cool, much less code.

**Review by guagualvcha — CHANGES_REQUESTED** [review]
Try find some performance friendly implementation.

File: core/statedb/statedb.go (hunk 1 above)
Comment (yutianwu) [reply core/statedb/statedb.go:197]: it seems the call is to query the account from the cache, we can not remove the call if we want to implement this function in this way

Comment (lightning-li) [reply core/statedb/statedb.go:197]: Yeah, I mean changing from `for accountIndex := range s.PendingAccountMap` to `for accountIndex, account := range s.PendingAccountMap`

Comment (KeefeL) [reply core/statedb/statedb.go]: Fixed, thanks.

[the remark this answers is not in the document]
Comment (KeefeL) [reply core/statedb/statedb.go]: Add comments, dedicated for the executor, not as generic API.

**Review by yutianwu — APPROVED** [review]

**Review by guagualvcha — APPROVED** [review]

**Pull request merged**